modules/contentpool_remote_register/src/PushManager.php

```diff
@@ -84,70 +85,66 @@ class PushManager implements PushManagerInterface, DestructableInterface {
    *   The messenger service.
    * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
    *   The config factory.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   The config factory service.
    */
-  public function __construct(EntityTypeManagerInterface $entity_type_manager, Serializer $serializer, ClientInterface $http_client, SensitiveDataTransformer $sensitive_data_transformer, MessengerInterface $messenger, ConfigFactoryInterface $config_factory) {
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, Serializer $serializer, ClientInterface $http_client, SensitiveDataTransformer $sensitive_data_transformer, MessengerInterface $messenger, ConfigFactoryInterface $config_factory, LoggerChannelFactoryInterface $logger_factory) {
     $this->entityTypeManager = $entity_type_manager;
     $this->httpClient = $http_client;
     $this->sensitiveDataTransformer = $sensitive_data_transformer;
     $this->messenger = $messenger;
     $this->serializer = $serializer;
     $this->configFactory = $config_factory;
+    $this->logger = $logger_factory->get('contentpool_remote_register');
   }
 
   /**
    * {@inheritdoc}
    */
   public function pushToRegisteredRemotes() {
```

Comment:
this is syncronous by default, but should be async, i.e. drupal should continue processing the request and not wait for all responses. I guess it would be enough to just remove the promise->wait() in the end?

Reply:
ah, I understand this method is synchron, while the other one can do both.